src/transformers/models/bert/modeling_bert.py

```diff
@@ -269,7 +269,8 @@ def forward(
             position_ids_l = torch.arange(seq_length, dtype=torch.long, device=hidden_states.device).view(-1, 1)
             position_ids_r = torch.arange(seq_length, dtype=torch.long, device=hidden_states.device).view(1, -1)
             distance = position_ids_l - position_ids_r
-            positional_embedding = self.distance_embedding(distance + self.max_position_embeddings - 1)
+            positional_embedding = self.distance_embedding(distance + self.max_position_embeddings).clip(-self.max_position_embeddings,
```

Comment:
Could you give a bit more context on this here? This is a big breaking change -> was it incorrect previously?

Reply:
Also pinging the author of the PR in question: https://github.com/huggingface/transformers/pull/8276, here: @zhiheng-huang .

Reply:
Yes, this is intentional.
Honestly, I'm not confident with this change because yes, it breaks backward compatibility.

Still, it makes sense to me to have 32 + 32 + 1 number of positions if the config has 32.
Istead of 31 + 31 +1.
Some other implementations have the same 2*pos_number + 1 approach.
Like this one
https://github.com/lucidrains/local-attention/blob/master/local_attention/local_attention.py#L95

If you insist, I should revert this of course.